integration-tests/chain-signatures/tests/actions/mod.rs

```diff
@@ -168,36 +265,36 @@ pub async fn single_payload_signature_production(
 
 #[tokio::test]
 async fn test_proposition() {
-    let big_r = "044bf886afee5a6844a25fa6831a01715e990d3d9e96b792a9da91cfbecbf8477cea57097a3db9fc1d4822afade3d1c4e6d66e99568147304ae34bcfa609d90a16";
-    let s = "1f871c67139f617409067ac8a7150481e3a5e2d8a9207ffdaad82098654e95cb";
-    let mpc_key = "02F2B55346FD5E4BFF1F06522561BDCD024CEA25D98A091197ACC04E22B3004DB2";
+    use k256::sha2::{Digest, Sha256};
 
-    // Create payload
     let mut payload = [0u8; 32];
     for (i, item) in payload.iter_mut().enumerate() {
         *item = i as u8;
     }
+    //
+    let big_r = "044bf886afee5a6844a25fa6831a01715e990d3d9e96b792a9da91cfbecbf8477cea57097a3db9fc1d4822afade3d1c4e6d66e99568147304ae34bcfa609d90a16";
+    let s = "1f871c67139f617409067ac8a7150481e3a5e2d8a9207ffdaad82098654e95cb";
+    let mpc_key = "02F2B55346FD5E4BFF1F06522561BDCD024CEA25D98A091197ACC04E22B3004DB2";
+    let account_id = "acc_mc.test.near";
+
+    let mut hasher = Sha256::new();
+    hasher.update(payload);
+    let mut payload_hash: [u8; 32] = hasher.finalize().into();
 
-    // TODO: get hashed values on the flight
-    let payload_hash: [u8; 32] = [
-        99, 13, 205, 41, 102, 196, 51, 102, 145, 18, 84, 72, 187, 178, 91, 79, 244, 18, 164, 156,
-        115, 45, 178, 200, 171, 193, 184, 88, 27, 215, 16, 221,
-    ];
-    let payload_hash_scalar = k256::Scalar::from_bytes(&payload_hash); // TODO: why do we need both reversed and not reversed versions?
-    let mut payload_hash_reversed: [u8; 32] = payload_hash;
-    payload_hash_reversed.reverse();
+    payload_hash.reverse();
```

Comment:
We are still reversing the hash here.

Reply:
This code was and still is a bit of a mess.

Previously converting a Scalar to bytes reversed the bytes and converted to a Scalar. 

The big_r and s values were generated using chain signatures from an older commit, therefore the signature is generated against a reversed hash.

You now have to reverse the bytes to generate the same Scalar and therefore Signature and this tests that. I'll write a comment to this effect, after this change has rolled out we should update this test using new values.